packages/grid/x-data-grid/src/components/cell/GridCell.tsx

```diff
@@ -125,16 +107,112 @@ const GridCell = React.forwardRef<HTMLDivElement, GridCellProps>((props, ref) =>
     ...other
   } = props;
 
-  const valueToRender = formattedValue == null ? value : formattedValue;
+  const field = column.field;
   const cellRef = React.useRef<HTMLDivElement>(null);
   const handleRef = useForkRef(ref, cellRef);
   const focusElementRef = React.useRef<FocusElement>(null);
   const apiRef = useGridApiContext();
-
   const rootProps = useGridRootProps();
+
+  const editRowsState = useGridSelector(apiRef, gridEditRowsStateSelector);
+
+  const { classes: rootClasses, getCellClassName } = rootProps;
+
+  const isSelected = useGridSelector(apiRef, () =>
+    apiRef.current.unstable_applyPipeProcessors('isCellSelected', false, {
+      id: rowId,
+      field,
+    }),
+  );
+
+  const cellParams = useGridSelector(
+    apiRef,
+    () => {
+      // This is required because `.getCellParams` tries to get the `state.rows.tree` entry
+      // associated with `rowId`/`fieldId`, but this selector runs after the state has been
+      // updated, while `rowId`/`fieldId` reference an entry in the old state.
+      try {
+        return apiRef.current.getCellParams<any, any, any, GridTreeNodeWithRender>(rowId, field);
+      } catch (e) {
+        if ((e as Error).message.startsWith('No row with id')) {
+          return null as any;
+        }
+        throw e;
+      }
+    },
+    shallowCompare,
+  );
+
+  const classNames = apiRef.current.unstable_applyPipeProcessors('cellClassName', [], {
+    id: rowId,
+    field,
+  });
+
+  if (column.cellClassName) {
+    classNames.push(
+      clsx(
+        typeof column.cellClassName === 'function'
+          ? column.cellClassName(cellParams)
+          : column.cellClassName,
+      ),
+    );
+  }
+
+  if (getCellClassName) {
+    classNames.push(getCellClassName(cellParams));
+  }
+
+  const editCellState = editRowsState[rowId]?.[column.field] ?? null;
+  const managesOwnFocus = column.type === 'actions';
+
+  const cellMode = cellParams.cellMode;
+  const hasFocus = cellParams.hasFocus;
+  const isEditable = cellParams.isEditable;
+  const valueToRender =
+    cellParams.formattedValue == null ? cellParams.value : cellParams.formattedValue;
+
   const ownerState = { align, showRightBorder, isEditable, classes: rootProps.classes, isSelected };
   const classes = useUtilityClasses(ownerState);
 
+  let children: React.ReactNode;
+  /* eslint-disable-next-line no-lone-blocks */
+  {
```

Comment:
This code block seems to be unnecessary

Reply:
Everything inside the block is initializing the `children`. I feel like it gives a bit of organization, and allows to fold that section of the code easily. I can remove it, I prefer it but I don't insist too much on it.